pkg/registry/resource/resourceclaim/declarative_validation_test.go

```diff
@@ -175,6 +175,41 @@ func testDeclarativeValidateUpdate(t *testing.T, apiVersion string) {
 			update: validClaim,
 			old:    validClaim,
 		},
+		"spec immutable: modify request class name": {
```

Comment:
Do we have a tests case for valid (unchanged) value?

Also, let's add a CREATE test, just to demonstrate that immutability doesn't affect create requests.

Reply:
> Do we have a tests case for valid (unchanged) value?
IIUC we do have a test case for this currently:
```go
		"valid": {
			update: validClaim,
			old:    validClaim,
		},
```

> Also, let's add a CREATE test, just to demonstrate that immutability doesn't affect create requests.
IIUC we have this test case already with:
```go
		"valid": {
			input: mkValidResourceClaim(),
		},
```

This test which creates a ResourceClaim with a fully populated Spec shows that the immutability check does not prevent the creation of the resource.